File: Exception/EventDelegation.py
# Python equivalent of event delegation via parent-level dispatch.

def task_list_click(event):
    target = event.get("target", {})
    if target.get("class") == "delete-btn":
        print("Delete item")
    elif target.get("class") == "complete-btn":
        print("Toggle done")


task_list_click({"target": {"class": "delete-btn"}})
task_list_click({"target": {"class": "complete-btn"}})

File: Exception/EventDelegation.js
// Instead of attaching listeners to each <li>, attach once to the parent <ul>
document.querySelector("#taskList").addEventListener("click", function(event) {
  const target = event.target;

  if (target.matches(".delete-btn")) {
    target.closest("li").remove();
  } else if (target.matches(".complete-btn")) {
    target.closest("li").classList.toggle("done");
  }
});